packages/runtime/container-runtime/src/test/blobManager.spec.ts

```diff
@@ -141,6 +153,7 @@ class MockRuntime
 	public attachState: AttachState;
 	public attachedStorage = new DedupeStorage();
 	public detachedStorage = new NonDedupeStorage();
+	public rejectStorage = new MockFailDedupeStorage();
```

Comment:
i'm not sure this works right, as we still want the return of createBlob to be something that doesn't synchronously return, and throwing immediatly won't really do that. I also think this class already has way to many options/flags, which make it hard to use and maintain

I think a better solution would be to have processBlobs take an error, or a flag, and allow that function to reject outstanding promises. I'm pretty sure if you rejected processBlobsP you would be most of the way to getting it to work.